codegens/curl/lib/index.js

```diff
@@ -4,6 +4,40 @@ var sanitize = require('./util').sanitize,
   _ = require('./lodash'),
   self;
 
+/**
+ *
+ * @param {*} urlObject The request sdk request.url object
+ * @returns {String} final url string converted from parsing url object
+ */
+function getUrlStringfromUrlObject (urlObject) {
+  var url = '';
+  if (urlObject.protocol) {
+    url += (urlObject.protocol.endsWith('://') ? urlObject.protocol : urlObject.protocol + '://');
+  }
+  if (urlObject.auth && urlObject.auth.user) {
+    url = url + ((urlObject.auth.password) ?
```

Comment:
@shreys7 Is this code copied from the postman-collection module? Any reason we can't use what's already there?

Reply:
Yes, this is the code copied from postman-collection module.
As we discussed, previously we were using `request.url.toString()` method which does all of this but **without encoding queryParams**. For this we stopped using that `.toString()` and copied the implementation of the same with just one tweak of setting encoding as true in `getQueryString({ encode: true })` method.